### DISCUSSION TITLE
Give Lambda a static IP

### DISCUSSION THREAD

**Discussion by mzepol07** [aws/aws-cdk#16506] [Q&A] [answered]
I need to give my lambda static IP so that it can be whitelisted by another service.  
I found a blog that lays out these steps, but I want to be able to do them in CDK (Python)
    Create a new VPC to run your code in (or use an existing VPC)
    Create a new Internet Gateway to communicate with the Internet from inside your VPC
    Create a Public Subnet and add a new route to the route table which routes to your Internet Gateway from 0.0.0.0/0
    Create a new Elastic IP address.
    Create a new NAT Gateway and assign it to the Public Subnet and Elastic IP address that you just created.
    Create a Private Subnet and add a new route to the route table which routes to your NAT Gateway from 0.0.0.0/0

I have an existing VPC and have created the Lambda function.  Then I created a private subnet using the ec2 class and added a nat gateway to it..
```
 # create a public subnet
public_subnet = ec2.PublicSubnet(
    self,
    id=f"external-lambda-subnet-{env_label}",
    vpc_id=vpc_id
)
nate_gateway = public_subnet.add_nat_gateway()
```
Here are my issues...
How do I create an internet gateway within the VPC?
How do I get back from nat gateway ID from the add_nat_gateway call?  I don't know the structure of that response body
How do I add routes to the public subnet within CDK? I imagine this will be a cfn fix.

Anyway, just kinda lost ovrall.  Let me know if you have any tips or examples.
Thanks.

**Comment by peterwoodworth** [comment, marked as answer]
Please view our [API ref on EC2](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2-readme.html), it should answer your questions. For example, the return type of addNatGateway will reveal the id from this prop https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.CfnNatGateway.html#attrnatgatewayid.

You can add routes to the subnet with the [addRoute method](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.PublicSubnet.html#addwbrrouteid-options)

Docs on internet gateway here https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2-readme.html#accessing-the-internet-gateway
